File: app/src/main/java/com/taianesilva/myshared/MainActivity.java
package com.taianesilva.myshared;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    EditText edittext_firstname, edittext_lastname, edittext_email;
    Button save_button;
    SharedPreferences sharedPrefences;

    private static final String SHARED_PREF_FIRSTNAME = "myfirstname";
    private static final String SHARED_PREF_LASTNAME = "mylastname";
    private static final String KEY_EMAIL = "email";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    edittext_firstname = findViewById(R.id.edittext_firstname);
    edittext_lastname = findViewById(R.id.edittext_lastname);
    edittext_email = findViewById(R.id.edittext_lastname);
    save_button = findViewById(R.id.save_button);

    sharedPrefences = getSharedPreferences(SHARED_PREF, MODE_PRIVATE);

    String firstname = sharedPrefences.getString(SHARED_PREF_FIRSTNAME,null);
    String lastname = sharedPrefences.getString(SHARED_PREF_LASTNAME,null);
    String email = sharedPrefences.getString(KEY_EMAIL, null);

    if (firstname != null || lastname != null || email != null){
            Intent intent = new Intent(MainActivity.this, HomeActivity.class);
            startActivity(intent);
        }

    save_button.setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View v) {
        SharedPreferences.Editor editor = sharedPrefences.edit();
        editor.putString(SHARED_PREF_FIRSTNAME,edittext_firstname.getText().toString());
        editor.putString(SHARED_PREF_LASTNAME,edittext_lastname.getText().toString());
        editor.putString(KEY_EMAIL,edittext_email.getText().toString());
        editor.apply();

            Intent intent = new Intent(MainActivity.this,HomeActivity.class);
            startActivity(intent);

            Toast.makeText(MainActivity.this, "Login Success", Toast.LENGTH_SHORT).show();
        }
    });

    }
}

File: app/src/main/java/com/taianesilva/myshared/MainActivity.kt
package com.taianesilva.myshared

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    lateinit var edittext_username:EditText
    lateinit var edit_password:EditText
    lateinit var edittext_newaccount:EditText
    lateinit var save_button:Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        edittext_username = findViewById(R.id.edittext_username)
        edit_password = findViewById(R.id.edit_password)
        edittext_newaccount = findViewById(R.id.edittext_newaccount)

        findViewById<Button>(R.id.save_button).setOnClickListener{
            saveData ()
        }
        findViewById<EditText>(R.id.edittext_newaccount).setOnClickListener{

        }
    }
    private fun saveData() {

        if(edittext_username != null) {
            println("Please enter with your username")
        }

        if(edit_password != null) {
            println("Please enter with your password")
        }
        return
       /* if(edittext_username.text.isEmpty()){
            edittext_username.error = "Please enter with your username"
            return
        }
        if(edit_password.text.isEmpty()){
            edit_password.error = "Please enter your password"
            return*/



        val mypref = getSharedPreferences("mypref", Context.MODE_PRIVATE)
        val editor = mypref.edit()

        editor.putString("Username", edittext_username.text.toString())
        editor.putString("Password", edit_password.text.toString())

        editor.apply()
        Toast.makeText(this, "Data saved", Toast.LENGTH_LONG).show()


    }


}